### DISCUSSION TITLE
Pre-Generation in Operator Signatures

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#126] [Features: On Hold]
```
( symbol typegen
 (int :sw ==> string :T)
 (
  (sw 1 ==) ("int" @T) when
  (sw 2 ==) ("flt" @T) when
  (sw 3 ==) ("str" @T) when
  (sw 4 ==) ("int|flt" @T) when
 ...
 )
) ::

3 :state

( symbol test
 (state typegen :x ==>)
 (x puts!)
) ::

10 test
```

Pre-generation in operator signatures would allow to use dynamic symbols. You may set a pre-generation in type-classes like generics. Generics and this idea seems no different at the main, i think.

What do you think about it?

**Comment by ghost** [comment]
That could solve #125

**Reply by h3rald** [reply]
Mmmm that would complicate operator signatures quite a bit! I'd rather not perform additional evaluation (no evaluation occurs in signatures, just lookups/validations)... things are probably already slow as it is! 😞

You could however create a lambda that generates a type class... 😉 you'd probably have to play a bit with scopes but it could work.

**Reply by ghost** [reply]
> you'd probably have to play a bit with scopes but it could work.

I am opening a portal here for scopes #117 
Seems there is going to be a discussion cobweb. 😄 

The _scope_ system should be improved for there a real usage of them, i say.

**Reply by ghost** [reply]
> things are probably already slow as it is! 😞

Hmm.. True but problem is the language isn't strong enough , you see. Somethings lack. So if you implement or change or fix these, there will be more useful at most cases.
